wandb/wandb_torch.py

```diff
@@ -160,9 +160,9 @@ def log_tensor_stats(self, tensor, name):
             cls = type(tensor)
             raise TypeError(f"Expected Tensor, not {cls.__module__}.{cls.__name__}")
 
-        # HalfTensors on cpu do not support view(), upconvert to 32bit
-        if isinstance(tensor, torch.HalfTensor):
-            tensor = tensor.clone().type(torch.FloatTensor).detach()
+        # most tensor types do not support many operations, convert to default
+        if tensor.dtype != torch.float32:
```

Comment:
If this is a torch.float64 tensor, we'd lose precision. Could you please instead list the specific uncommon types that require this, in addition to HalfTensor?